src/compiler/checker.ts

```diff
@@ -7167,6 +7176,16 @@ namespace ts {
 
             getNodeLinks(node).flags |= nodeCheckFlag;
 
+            // Due to how we emit async functions, we need to specialize the emit for an async method that contains a `super` reference.
+            if (container.kind === SyntaxKind.MethodDeclaration && container.flags & NodeFlags.Async) {
+                if ((isSuperPropertyAccess(node.parent) || isSuperElementAccess(node.parent)) && isAssignmentTarget(node.parent)) {
+                    getNodeLinks(container).flags |= NodeCheckFlags.AsyncMethodWithSuperBinding;

```

Comment:
Add a comment including example of the case where flags is `NodeCheckFlags.AsyncMethodWithSuperBinding`